### PR TITLE
Tasks/samples.harness

### PR DESCRIPTION
**Pull request by cinlyooi-intel**
This pull request improve the ability for test framework to comprehend and confirm that test runs on two sample tests do cover all important areas in the test cases. With a more verbose output, it is also easier to identify which part of the test had failed.

This PR was developed when it was noted that the two test cases in samples/subsys/usb/cdc_acm did not produce the correct result in a few circumstances: 

(1) It was demonstrated that if the test run got stuck between  "Waiting for DTR" and "DTR set, start test", the testcase "usb.cdc-acm" incorrectly passed because it was only looking for the string "Wait for DTR".

(2) testcase "usb.cdc-acm_comp" was setup wrongly. It was setup to detect the end of the test case but the last "while(1)" loop had ensured that in the test run will not exit and the test case will not end. If the test run did exit, it is actually a failure.

samples/subsys/usb/cdc_acm_composite exibit similar behaviour as samples/subsys/usb/cdc_acm and was updated.

### CODE REVIEW

File: samples/subsys/usb/cdc_acm/src/main.c [hunk 1]
```diff
@@ -118,7 +118,12 @@ void main(void)
 	uart_irq_rx_enable(dev);
 
 	/* Echo the received data */
+	int i = 0;
```
Comment (finikorg) [inline samples/subsys/usb/cdc_acm/src/main.c]: I think you should put declaration in the beginning of the function, or use `for()` loop declaring it.

**Review by finikorg — CHANGES_REQUESTED** [review]
BTW, how can I run those harness tests?

**Comment by finikorg** [comment]
You should also rename commits to be more readable, now they have similar name.
`tests: cdc_acm: *****` etc

File: samples/subsys/usb/cdc_acm/src/main.c (hunk 1 above)
Comment (cinlyooi-intel) [reply samples/subsys/usb/cdc_acm/src/main.c]: Thank you for the reminder. 
Done as requested. I moved declaration to top

**Comment by cinlyooi-intel** [comment]
> You should also rename commits to be more readable, now they have similar name.
> `tests: usb: cdc_acm: *****` etc

Done as requested.

File: samples/subsys/usb/cdc_acm/sample.yaml [hunk 2]
```diff
@@ -6,10 +6,23 @@ tests:
...
     harness: console
     harness_config:
-      type: one_line
+      type: multi_line
       regex:
         - "Wait for DTR"
+        - "DTR set, start test"
+        - "Baudrate detected"
+        - "Starting indefinite read/write data loop"
+        - "At least one read/write data cycle completed"
```
Comment (finikorg) [inline samples/subsys/usb/cdc_acm/sample.yaml]: Have toy tested it working? I have following output:
```
***** Booting Zephyr OS v1.14.0-rc1-447-ga40f5a5c1a *****
Wait for DTR
DTR set, start test
Baudrate detected: 115200
```

File: samples/subsys/usb/cdc_acm/src/main.c [hunk 3]
```diff
@@ -120,5 +121,9 @@ void main(void)
 	/* Echo the received data */
 	while (1) {
 		read_and_echo_data(dev, (int *) &bytes_read);
+		i++;
+		if (i == 1) {
+		    printf("At least one read/write data cycle completed");
```
Comment (finikorg) [inline samples/subsys/usb/cdc_acm/src/main.c]: I do not think this is needed, can sanitycheck inject some data?

**Review by finikorg — CHANGES_REQUESTED** [review]

File: samples/subsys/usb/cdc_acm/src/main.c (hunk 3 above)
Comment (finikorg) [reply samples/subsys/usb/cdc_acm/src/main.c]: I tested with following cmd:
```
sanitycheck --device-testing --device-serial /dev/ttyACM0 -p reel_board -T samples/subsys/usb/cdc_acm
```

File: samples/subsys/usb/cdc_acm/sample.yaml (hunk 2 above)
Comment (finikorg) [reply samples/subsys/usb/cdc_acm/sample.yaml]: @nashif For this to work some extra steps needs to be done:
- we need to open another console
- send characters to another console
Do we have command in sanitycheck to open another console and inject characters?

**Comment by cinlyooi-intel** [comment]
After reviewing the PR with @finikorg, it is clear that the existing implementation is less likely to cause test automation to report false positive. As such, this PR is withdrawn

**Pull request closed without merging**